packages/plugin-ext/src/common/plugin-api-rpc.ts

```diff
@@ -1424,6 +1425,38 @@ export interface WebviewsMain {
     $unregisterSerializer(viewType: string): void;
 }
 
+export interface CustomEditorsExt {
+    $resolveCustomEditorWebview(
+        handle: string,
+        viewType: string,
+        resource: URI,
+        title: string,
+        options: theia.WebviewPanelOptions,
+        cancellation: CancellationToken): Promise<void>;
+    $createCustomDocument(resource: UriComponents, viewType: string, backupId: string | undefined, cancellation: CancellationToken): Promise<{ editable: boolean }>;
+    $disposeCustomDocument(resource: UriComponents, viewType: string): Promise<void>;
+    $undo(resource: UriComponents, viewType: string, editId: number, isDirty: boolean): Promise<void>;
+    $redo(resource: UriComponents, viewType: string, editId: number, isDirty: boolean): Promise<void>;
+    $revert(resource: UriComponents, viewType: string, cancellation: CancellationToken): Promise<void>;
+    $disposeEdits(resourceComponents: UriComponents, viewType: string, editIds: number[]): void;
+    $onSave(resource: UriComponents, viewType: string, cancellation: CancellationToken): Promise<void>;
+    $onSaveAs(resource: UriComponents, viewType: string, targetResource: UriComponents, cancellation: CancellationToken): Promise<void>;
+    $onMoveCustomEditor(handle: string, newResource: UriComponents, viewType: string): Promise<void>;
+}
```

Comment:
I didn't start this PR so had to check and after checking and starting implementing the backup functionality I came to conclusion that backup should be done in separate PR because the infrastructure for backup doesn't exist in Theia and this isn't a small addition - it adds many interfaces and files for current PR. It also uses URI.joinPath - which needs upgrade of monaco editor to 0.21x - although it is possible to temporarily use a local function,  and uses also a Hash module.

Reply:
I see. Thank you for the explanation!